src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionDispatcher.cs

```diff
@@ -505,6 +505,15 @@ private static bool ServerHasWebSockets(IFeatureCollection features)
                         context.Response.ContentType = "text/plain";
                         return;
                     }
+                    catch (IOException ex)
+                    {
+                        // Can occur when the HTTP request is canceled by the client
+                        Log.BadHttpRequest(_logger, connection.ConnectionId, ex);
+
+                        context.Response.StatusCode = StatusCodes.Status400BadRequest;
```

Comment:
The interesting thing about this is that the client is already gone and that's why you got an IOException happened in the first place. The chances of the response going somewhere is 0

Reply:
It occurred to me when I wrote it. Doesn't hurt 🤷‍♂️

Reply:
I'd remove it.

Reply:
Can there be an IOException from Kestrel for something other than a FIN? It could be seen in that case.

Reply:
a fin usually results in a graceful close (steam returns 0). I guess setting headers doesn’t hurt as long as we don’t explicitly try to write anything.